staker-rewards/README.adoc

```diff
@@ -1,35 +1,108 @@
+rewards calculations are performed weekly off-chain, and committed for
+distribution using a variant of Uniswap’s Merkle distributor This repository
+contains scripts used to calculate rewards and generate Merkle tree used by
+stakers to claim their rewards. Anyone can use tho scripts to validate reward
+distribution or to generate reward estimates for a reward interval still
+pending.
+
+== Algorithm
+
+The algorithm for calculating ECDSA staker rewards as implemented in the scripts 
+in this repository is described in this section.
+
+=== Input
+* Interval `T` starts at `T_start`.
+
+* Interval `T` ends at `T_end`.
+
+* All operators who have ever bonded ETH are taken into account.
+
+=== Operator Parameters
+For each Operator:
+
+* Calculate the amount of KEEP staked at `T_end` as `KEEP_staked`.
+
+* Calculate the total amount of ETH bonded at `T_start` as `ETH_bonded`.
+
+* Read the amount of ETH unbonded at `T_start` as `ETH_unbonded`.
+
+* Calculate the sum of ETH unbonded and withdrawn for the entire interval
+  as `ETH_withdrawn`.
+  
+* Calculate the total ETH under management as 
+  `ETH_total = ETH_bonded + ETH_unbonded - ETH_withdrawn`.
+
+=== Requirements
+
+* If the operator does not have `BondedECDSAKeepFactory` contract or tBTC sortition
+  pool authorized at `T_start`, they are not getting any rewards for interval `T`.
+  
+* If the operator does not have a minimum KEEP stake at `T_start`, they are not
+  getting any rewards for interval `T`.
+  
+* If the operator has more `ETH_unbonded` than the minimum unbonded value and is
+  not in the sortition pool at `T_start`, they are not getting any rewards for
+  interval `T`. In other words, if the operator is eligible to be in the
+  sortition pool, they have to be in the sortition pool. Operators who are not
+  in the sortition pool because all of their ETH is bonded are still getting
+  rewards because that ETH is still under the system’s management.
+  
+* If the operator deauthorized tBTC sortition pool between `T_start` and `T_end`, 
+  they are not getting any rewards for interval `T`.
+  
+* If the operator ever committed a fraud, they are not getting any rewards for
+  interval `T`. 
+  
+* If the operator did not meet the signature SLA between `T_start` and `T_end`,
+  they are not getting any rewards for interval `T`. For 20 or less
+  closed/terminated keeps, no more than one could be terminated with a signature
+  timeout. For more than 20 closed/terminated keeps no more than 5% could be
+  terminated with a signature timeout. Signature SLA does not include not
+  answered courtesy calls.
+  
+* If the operator did not meet the key generation SLA between `T_start` and
+  `T_end` , they are not getting any rewards for interval `T` . For less than
+  10 key generations, one of them can fail. F For more than 10 key generations,
+  no more than 10% can fail.
+  
+=== The Formula
+
+image::rewards.png[staker rewards formula]
```

Comment:
This formula has a slightly different naming than used in the script. For example `ethLocked` is actually the `ethTotal` parameter. It may be worth putting a note about this.

Reply:
I think I'd rather update the script. At least column headers. `ETH_locked` makes more sense IMO than `ETH_total`. WDYT?

Reply:
I have also done a rename in the Operator Parameters section to stay consistent in the README. Please see https://github.com/keep-network/keep-ecdsa/pull/642/commits/46d2a07968d0954dfa5c53f0d831c6a15f58bfd1.

Reply:
Yep, I think we can rename it in the script.

Reply:
OK, let's leave it for a separate PR then. It's not a blocker right now.

Reply:
Ok, sure.